Libraries/Opc.Ua.Client/Subscription.cs

```diff
@@ -1958,15 +2006,26 @@ private async Task OnMessageReceived()
                             m_messageCache.AddLast(ii.Value.Message);
                             ii.Value.Processed = true;
 
-                            //Keep the last sequence number processed going up
+                            // Keep the last sequence number processed going up
                             if (ii.Value.SequenceNumber > m_lastSequenceNumberProcessed)
                             {
                                 m_lastSequenceNumberProcessed = ii.Value.SequenceNumber;
+                                m_resyncLastSequenceNumberProcessed = false;
```

Comment:
I vaguely remember that `m_incomingMessages` is populated in sequence-number order. (It's ordered by it) so the first entry to hit up this branch turning off the flag is fine because the subsequent messages should now obey the sequence rules imposed by `ValidSequentialPublishMessage`

Just confirming, if that's the case then it looks fine to me

Reply:
I recall seeing more areas of the code not support overflow, so I didn't add handling for it either. 

Overflow of 4294967295, given one millisecond per message, that's just under half a year of uptime.

Reply:
actually 10ms is sampling, publish is still limited to 50ms...

Reply:
Right. I recalled 50ms as some limit but didn't want to assume. 
At 50ms per publish, it'll overflow no sooner than 6.8 years... 

If our system has that kind of uptime, we should be so lucky to suffer from success.

I can appreciate how that might be an issue for others though. The logic to handle it is simple enough in theory, but it needs to be done in a bunch of places regardless of sequential publishing. For example, the order it inserts into the linked list.